python/lsst/ts/ess/common/sensor/campbell_csat3b.py

```diff
@@ -0,0 +1,170 @@
+# This file is part of ts_ess_common.
+#
+# Developed for the Vera C. Rubin Observatory Telescope and Site Systems.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["compute_signature", "Csat3bSensor"]
+
+import logging
+import numpy as np
+
+from .base_sensor import BaseSensor
+from ..constants import SensorType
+from .sensor_registry import register_sensor
+
+_NUM_VALUES = 7
+
+
+def compute_signature(input_str: str, delimiter: str) -> int:
+    """Compute the signature of the sensor telemetry.
+
+    The computation is based on the C function that can be found on page
+    55 of csat3b.pdf in the doc directory.
+
+    Parameters
+    ----------
+    input_str : `str`
+        The string of telemetry for which to compute the signature.
+    delimiter: `str`
+        The delimiter used in input_str.
+
+    Returns
+    -------
+    `int`
+        The signature.
+
+    Notes
+    -----
+    By experiment it was found that the description of the input string on
+    page 55 of the document is incorrect. The description states that all
+    data from "x" to "c" are used but by implementing the C function in a C
+    source file and calling the function it was established that only "x"
+    up to "d" are used.
+    For compatibility sake, the input string is expected to contain the
+    value of "c" plus a leading delimiter and this Python implementation
+    strips that value and delimiter before computing the signature.
+    """
+    # Remove the value of "c" plus the leading delimiter from the input
+    # string otherwise an incorrect signature is computed.
+    last_index = input_str.rfind(delimiter)
+    input_str = input_str[:last_index]
+
+    # This is a Python version of the C function.
+    seed = 0xAAAA
+    msb = seed >> 8
+    # Ensure that the value of lsb is in the range of a C unsigned short.
+    lsb = seed % 256
```

Comment:
Please consider using 0x100 instead of 256 here and below.

Even if the manual uses 256 I'd be tempted to make that change. I feel that using decimal constants for bit operations (except when shifting bits) makes the code harder to understand. Though at least 256 is a well known value.

Reply:
Done